FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+ */
+static MQTTStatus_t prvProcessCommand( Command_t * pxCommand );
+
+/**
+ * @brief Dispatch incoming publishes and acks to response queues and
+ * command callbacks.
+ *
+ * @param[in] pMqttContext MQTT Context
+ * @param[in] pPacketInfo Pointer to incoming packet.
+ * @param[in] pDeserializedInfo Pointer to deserialized information from
+ * the incoming packet.
+ */
+static void prvSubscriptionManager( MQTTContext_t * pMqttContext,
+                                    MQTTPacketInfo_t * pPacketInfo,
+                                    uint16_t packetIdentifier,
+                                    MQTTPublishInfo_t * pPublishInfo );
+
+/**
+ * @brief Process commands from the command queue in a loop.
+ *
+ * This demo requires a process loop command to be enqueued before calling this
+ * function, and will re-add a process loop command every time one is processed.
+ * This demo will exit the loop after receiving an unsubscribe operation.
+ */
+static void prvCommandLoop();
+
+/**
+ * @brief Common callback for commands in this demo.
+ *
+ * This callback marks the command as complete and notifies the calling task.
+ *
+ * @param[in] pxContext Context of the initial command.
+ */
+static void prvCommandCallback( CommandContext_t * pxContext );
+
+/**
+ * @brief The task used to create various publish operations.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+void prvPublishTask( void * pvParameters );
+
+/**
+ * @brief The task used to wait for incoming publishes.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+void prvSubscribeTask( void * pvParameters );
+
+/**
+ * @brief The main task used in the MQTT demo.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+static void prvMQTTDemoTask( void * pvParameters );
+
+/**
+ * @brief The timer query function provided to the MQTT context.
+ *
+ * @return Time in milliseconds.
+ */
+static uint32_t prvGetTimeMs( void );
+
+/*-----------------------------------------------------------*/
+
+/* Global MQTT context. */
+static MQTTContext_t globalMqttContext;
+
+/* List of operations that are awaiting an ack from the broker. */
+static AckInfo_t pxPendingAcks[ PENDING_ACKS_MAX_SIZE ];
+
+/* List of active subscriptions. */
+static SubscriptionElement_t pxSubscriptions[ SUBSCRIPTIONS_MAX_COUNT ];
+
+/**
+ * @brief Queue for main task to handle MQTT operations.
+ */
+static QueueHandle_t xCommandQueue;
+
+/**
+ * @brief Response queue for prvPublishTask.
+ */
+static QueueHandle_t xResponseQueue1;
+
+/**
+ * @brief Response queue for prvSubscribeTask.
+ */
+static QueueHandle_t xResponseQueue2;
+
+/**
+ * @brief Handle for prvMQTTDemoTask.
+ */
+static TaskHandle_t xMainTask;
+
+/**
+ * @brief Handle for prvPublishTask.
+ */
+static TaskHandle_t xTask1;
```

Comment:
A name to indicate which task would be better here.

Reply:
Renamed to xPublisherTask